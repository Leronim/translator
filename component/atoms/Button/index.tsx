import React, { useState } from 'react';
import { Pressable, StyleSheet, Text, Dimensions } from 'react-native';
import { Neomorph } from 'react-native-neomorph-shadows';

interface Props {
    text: string,
    onPress: () => void
}

const CheckButton: React.FC<Props> = ({ text, onPress }) => {
    const [ swap, setSwap ] = useState<boolean>(false);

    return (
        <Neomorph
            inner={swap}
            darkShadowColor='#000'
            lightShadowColor='#fff'
            style={styles.neomorph}
        >
            <Pressable 
                onPressIn={() => setSwap(true)} 
                onPressOut={() => setSwap(false)}
                style={styles.screenContainer} 
                onPress={() => onPress()}
            >
                <Text style={styles.text}>{text}</Text>
            </Pressable>
        </Neomorph>
    )
}

const styles = StyleSheet.create({
    screenContainer: {
        height: '100%',
        justifyContent: 'center',
        alignItems: 'center'
    },
    neomorph: {
        shadowOpacity: 0.2,
        shadowRadius: 8,
        shadowOffset: {
            width: 2,
            height: 4
        },
        borderRadius: 88,
        backgroundColor: '#F2F2F2',
        height: 68,
        width: Dimensions.get('window').width - 40
    },
    text: {
        fontStyle: 'normal',
        fontWeight: 'bold',
        fontSize: 18,
        lineHeight: 21,
        color: "#000"
    }
})

export default CheckButton;