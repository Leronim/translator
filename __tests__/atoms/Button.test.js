import React from 'react';
import Button from '../../component/atoms/Button';
import { render, fireEvent } from '@testing-library/react-native';

test('Button render', () => {
    const tree = render(<Button text="Check"/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('click button', () => {
    const handleClick = jest.fn();

    const { getByTestId } = render(<Button text="Check" onPress={handleClick}/>);
    fireEvent.press(getByTestId("sendButton"));

    expect(handleClick).toHaveBeenCalled();    
})