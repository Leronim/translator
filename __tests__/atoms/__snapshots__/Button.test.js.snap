// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render 1`] = `
<View
  style={
    Object {
      "backfaceVisibility": undefined,
      "borderRadius": 34,
      "bottom": undefined,
      "direction": undefined,
      "display": undefined,
      "end": undefined,
      "height": 68,
      "left": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginEnd": undefined,
      "marginHorizontal": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginStart": undefined,
      "marginTop": undefined,
      "marginVertical": undefined,
      "opacity": undefined,
      "position": undefined,
      "right": undefined,
      "start": undefined,
      "top": undefined,
      "transform": undefined,
      "width": 710,
      "zIndex": undefined,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F2F2F2",
          "borderRadius": 34,
          "height": 68,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "x": 2,
            "y": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 16,
          "width": 710,
        },
        Object {
          "position": "absolute",
          "shadowOffset": Object {
            "height": 4,
            "width": 2,
          },
          "shadowRadius": 8,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F2F2F2",
          "borderRadius": 34,
          "height": 68,
          "shadowColor": "#fff",
          "shadowOffset": Object {
            "x": -2,
            "y": -4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 16,
          "width": 710,
        },
        Object {
          "position": "absolute",
          "shadowOffset": Object {
            "height": -4,
            "width": -2,
          },
          "shadowRadius": 8,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#F2F2F2",
        "borderRadius": 34,
        "height": 68,
        "width": 710,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
        }
      }
      testID="sendButton"
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": 21,
          }
        }
      >
        Check
      </Text>
    </View>
  </View>
</View>
`;
